### PR TITLE
Feature/systick

### PR DESCRIPTION
**Pull request by hmightypirate**

### CODE REVIEW

File: src/hal/setup.c [hunk 1]
```diff
@@ -130,6 +130,7 @@ static void setup_exceptions(void)
 {
 
   nvic_set_priority(NVIC_TIM1_UP_TIM10_IRQ, 0);
+  nvic_set_priority(NVIC_SYSTICK_IRQ, 1);  // systick interruptions are enabled in main (original bulebule code)
```
Comment (eLeDe) [inline src/hal/setup.c]: Priority levels only use bits [7:4].

File: src/main.c [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
  * @brief Empty main
  */
 int main(void)
 {
+
+  setup();
+
+  // FIXME: kinematic configuration goes here
+  
+  systick_interrupt_enable();
```
Comment (eLeDe) [inline src/main.c:13]: This function should go inside setup()

**Review by eLeDe — CHANGES_REQUESTED** [review]
Minor changes are needed

File: src/main.c (hunk 2 above)
Comment (punkto) [reply src/main.c:13]: Currently, yes, but later on we will need to do some things between setup and enabling the systick.

Maybe makes sense enable it on setup and disabling it inmediatly back in main if this way setup does performs all the functionality intended...

Comment (punkto) [reply src/main.c:13]: Anyway, this function shouldn't be here, as it  is defined in libopencm3, not in HAL.

**Comment by punkto** [comment]
sys_tick_handler should be implemented (in hal/systick.c ?)

**Review by eLeDe — COMMENTED** [review]
:ok_hand:

**Pull request merged**